assets/blocks/course-outline/store.js

```diff
@@ -38,13 +38,19 @@ const actions = {
 				data: { structure: yield getEditorStructure() },
 			} );
 			yield actions.setStructure( result );
+
+			yield dispatch( 'core/notices' ).removeNotice(
```

Comment:
There is a case where the message isn't removed: when adding an empty module, saving (error), then removing the module and saving again. 
This comes from the optimization that structure at this point is considered unchanged and save structure is not called. I don't think there is a feasible way to fix this as removing the message just on the editor `isSaving` state will cause issues.

Reply:
Good catch! It's tricky! I'm trying to find a solution for that.

Reply:
I refactored a little the saving to have a little more control. In summary, I added a new `saveCalled` to our store that flags whether the save was already called for that post saving.